modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/maven/aether/KobaltMavenResolver.kt

```diff
@@ -38,6 +38,14 @@ class KobaltMavenResolver @Inject constructor(val settings: KobaltSettings,
     fun resolve(id: String, scope: Scope? = null, filter: DependencyFilter? = null): DependencyResult {
         val dependencyRequest = DependencyRequest(createCollectRequest(id, scope), filter)
         val result = system.resolveDependencies(session, dependencyRequest)
+        if (args.downloadSources) {
+            listOf("sources", "javadoc").forEach {
+                val artifact = DefaultArtifact(id)
+                val sourceArtifact = DefaultArtifact(artifact.groupId, artifact.artifactId, it, artifact.extension,
+                        artifact.version)
+                system.resolveDependencies(session, DependencyRequest(createCollectRequest(sourceArtifact, scope), filter))
```

Comment:
In case if you want exactly one resolution in MavenResolver you should somehow filter session.resolveDependencies(...) DependencyResult and exclude sources/javadoc jars. So probably several separate resolutions in MavenResolver make sense....